### DISCUSSION TITLE
Android 12

### DISCUSSION THREAD

**Discussion by onley** [mrrfv/open-android-backup#39] [General]
Does this still work under Android 12? I have tried to perform a backup twice, once using tar export and once using adb export and both times it failed at "Exporting contacts (as vCard)". When using adb export there was also an error message 

adb: error: failed to stat remote object '/storage/emulated/0/linux-android-backup-temp/.': No such file or directory

apparently the companion app is unable to write to the device.

**Comment by mrrfv** [comment]
It should work for Android 12. Does the app ask you for "all files access" permissions after pressing the export data button?

**Reply by onley** [reply]
No, after pressing export on the phone the backup starts. it backs up all the apk files but then throws the error on the contacts.

**Comment by wgITos** [comment]
I observe the same issue for CalyxOS 4.3.0 (Android 13).

**Reply by mrrfv** [reply]
I tested the app & script on vanilla Android 13, and it worked just fine. Do you know if CalyxOS has some additional privacy features that prevent apps from directly modifying the internal storage? GrapheneOS has a feature like this.

**Reply by wgITos** [reply]
Not that I'm aware of. But I remember now that I could not edit the ASCII-profiles of the brouter app directly on the cell phone, or more precisely, I could save the changed file, but the file remained unchanged. I then copied the few files with adb to the Linux PC, edited them and moved them back.

**Comment by kooltux** [comment]
Backup done on zenfone9/android12 - works well for me.

**Comment by mrrfv** [comment]
I took multiple backups on my test device running pure Android 12, and the app worked perfectly. I'll try to debug this further with an Android 13 GSI tomorrow.

Edit: Works on TrebleDroid's vanilla Android 13 [build](https://github.com/TrebleDroid/treble_experimentations/releases/tag/ci-20221219), not sure why this doesn't work properly on all devices. @onley do you use a security-focused Android distribution like CalyxOS or GrapheneOS?

**Comment by mrrfv** [comment]
Please report further problems to #55 - this sounds like an issue with Calyx and GrapheneOS, which would explain why I'm unable to reproduce any of this on AOSP.
